src/components/ValidateCodeActionModal/index.tsx

```diff
@@ -38,13 +38,13 @@ function ValidateCodeActionModal({
     }, [onClose, clearError]);
 
     useEffect(() => {
-        if (!firstRenderRef.current || !isVisible) {
+        if (!firstRenderRef.current || !isVisible || hasMagicCodeBeenSent) {
             return;
         }
         firstRenderRef.current = false;
 
         sendValidateCode();
-    }, [isVisible, sendValidateCode]);
+    }, [isVisible, sendValidateCode, hasMagicCodeBeenSent]);
```

Comment:
Adding this dependency caused https://github.com/Expensify/App/issues/54046. The reasoning is explained in detail [here](https://github.com/Expensify/App/issues/54046#issuecomment-2539773965)